pkg/tasks/prometheusadapter.go

```diff
@@ -130,7 +130,27 @@ func (t *PrometheusAdapterTask) Run() error {
 		}
 	}
 	{
-		dep, err := t.factory.PrometheusAdapterDeployment()
+		apiAuthConfigmap, err := t.client.GetAPIServerAuthenticationConfigmap()
+		if err != nil {
+			return errors.Wrap(err, "failed to load kube-system/extension-apiserver-authentication configmap")
+		}
+
+		apiAuthSecret, err := t.factory.PrometheusAdapterAPIAuthSecret(apiAuthConfigmap.Data)
+		if err != nil {
+			return errors.Wrap(err, "initializing PrometheusAdapter Deployment failed")
```

Comment:
```suggestion
			return errors.Wrap(err, "failed to create prometheus adapter api auth secret")
```

Something along the lines?

Reply:
sounds good :+1: